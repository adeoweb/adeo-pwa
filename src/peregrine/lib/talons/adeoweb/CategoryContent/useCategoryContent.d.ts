import { DocumentNode } from 'graphql';

import {
    ProductSortDirections,
    ProductSortFields
} from 'src/lib/constants/product';
import { TAggregation } from 'src/lib/types/graphql/Aggregation';
import { TCategoryInterface } from 'src/lib/types/graphql/Category';
import { TProduct } from 'src/lib/types/graphql/Product';

type TUseCategoryContentProps = {
    query: DocumentNode;
    currentPage: number;
    categoryId: number;
    sortField: ProductSortFields;
    sortDir: ProductSortDirections;
    setTotalPages: (count: number) => void;
};

type TUseCategoryContent = {
    error: null | Error;
    loading: boolean;
    category: TCategoryInterface | null;
    products: TProduct[];
    filters: TAggregation[];
    totalCount: number;
    activeFilters: Map<string, string[]>;
    setFilter: (attributeCode: string, values: string[]) => void;
    toggleFilter: (attributeCode: string, value: string) => void;
    pageSize: number;
    setPageSize: (pageSize: number) => void;
};

export function useCategoryContent(
    props: TUseCategoryContentProps
): TUseCategoryContent;
