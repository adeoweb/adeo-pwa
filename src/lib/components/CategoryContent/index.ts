export { default } from './CategoryContent';
export {
    ICategoryPageControl,
    ICategoryContentProps
} from './CategoryContentTypes';
