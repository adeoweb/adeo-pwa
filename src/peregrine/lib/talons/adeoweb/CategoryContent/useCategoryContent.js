import { useLazyQuery } from '@apollo/react-hooks';
import { useCallback, useEffect, useState } from 'react';
import { useProductFilters } from 'src/peregrine/lib/talons/adeoweb/Product/useProductFilters';
import { usePageSize } from 'src/peregrine/lib/talons/adeoweb/Product/usePageSize';
import { fetchPolicy } from 'src/peregrine/lib/util/adeoweb/fetchPolicy';

const CATEGORY_ATTRIBUTE_CODE = 'category_id';

export const useCategoryContent = props => {
    const {
        query,
        currentPage,
        categoryId,
        sortField,
        sortDir,
        setTotalPages
    } = props;

    const [activeCategory, setActiveCategory] = useState(null);
    const {
        getFilterQuery,
        activeFilters,
        setFilter,
        toggleFilter,
        setActiveFilters
    } = useProductFilters();

    const handleToggleFilter = useCallback(
        (attributeCode, value) => {
            if (attributeCode === CATEGORY_ATTRIBUTE_CODE) {
                const values = activeFilters.get(attributeCode) || [];
                const valueIndex = values.indexOf(value);
                const newValues = [];

                if (valueIndex === -1) {
                    newValues.push(value);
                } else {
                    newValues.push(activeCategory.toString());
                }

                setFilter(attributeCode, newValues);
            } else {
                toggleFilter(attributeCode, value);
            }
        },
        [toggleFilter, setFilter, activeFilters, activeCategory]
    );

    const { pageSize, setPageSize } = usePageSize();
    const [data, setData] = useState(null);
    const [runQuery, queryResponse] = useLazyQuery(query, {
        fetchPolicy: fetchPolicy.queries.default,
        onCompleted: data => {
            setData(data);
        }
    });

    // Run the category query immediately and whenever its variable values change.
    useEffect(() => {
        runQuery({
            variables: {
                currentPage: Number(currentPage),
                id: String(activeCategory),
                onServer: false,
                pageSize: Number(pageSize),
                sort: {
                    [sortField]: sortDir
                },
                filter: getFilterQuery()
            }
        });
    }, [
        currentPage,
        activeCategory,
        pageSize,
        sortField,
        sortDir,
        runQuery,
        getFilterQuery,
        activeFilters,
        setActiveFilters
    ]);

    const { loading, error } = queryResponse;

    let category = null;
    if (data && data.categoryList && Array.isArray(data.categoryList)) {
        const [firstCategory] = data.categoryList;
        category = firstCategory;
    }

    let products = [];
    if (data && data.products && Array.isArray(data.products.items)) {
        products = data.products.items;
    }

    let filters = [];
    if (data && data.products && Array.isArray(data.products.aggregations)) {
        filters = data.products.aggregations;
    }

    let totalPages = 0;
    if (
        data &&
        data.products &&
        data.products.page_info &&
        data.products.page_info.total_pages
    ) {
        totalPages = data.products.page_info.total_pages;
    }

    let totalCount = 0;
    if (data && data.products && data.products.total_count) {
        totalCount = data.products.total_count;
    }

    useEffect(() => {
        setTotalPages(totalPages);
    }, [setTotalPages, totalPages]);

    useEffect(() => {
        setActiveCategory(categoryId);
        setActiveFilters(
            new Map([[CATEGORY_ATTRIBUTE_CODE, [categoryId.toString()]]])
        );
    }, [categoryId, setActiveCategory, setActiveFilters]);

    return {
        loading,
        error,
        category,
        products,
        filters,
        totalCount,
        activeFilters,
        setFilter,
        toggleFilter: handleToggleFilter,
        pageSize,
        setPageSize
    };
};
