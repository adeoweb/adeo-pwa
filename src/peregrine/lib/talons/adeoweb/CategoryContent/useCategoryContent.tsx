import { DocumentNode } from 'graphql';

import { useLazyQuery } from '@apollo/react-hooks';
import { useCallback, useEffect, useState } from 'react';

import {
    ProductSortDirections,
    ProductSortFields
} from 'src/lib/constants/product';
import {
    GetCategoryContentQuery,
    GetCategoryContentQueryVariables
} from 'src/lib/queries/getCategoryContent.generated';
import { TAggregation } from 'src/lib/types/graphql/Aggregation';
import { TCategoryInterface } from 'src/lib/types/graphql/Category';
import { TProduct } from 'src/lib/types/graphql/Product';
import { usePageSize } from 'src/peregrine/lib/talons/adeoweb/Product/usePageSize';
import { useProductFilters } from 'src/peregrine/lib/talons/adeoweb/Product/useProductFilters';
import { fetchPolicy } from 'src/peregrine/lib/util/adeoweb/fetchPolicy';
import filterOutNullableValues from 'src/peregrine/lib/util/adeoweb/filterOutNullableValues';

const CATEGORY_ATTRIBUTE_CODE = 'category_id';

type TUseCategoryContentProps = {
    query: DocumentNode;
    currentPage: number;
    categoryId: number;
    sortField: ProductSortFields;
    sortDir: ProductSortDirections;
    setTotalPages: (count: number) => void;
};

type TUseCategoryContent = {
    error?: string;
    loading: boolean;
    category: TCategoryInterface | null;
    products: TProduct[];
    filters: TAggregation[];
    totalCount: number;
    activeFilters: Map<string, string[]>;
    setFilter: (attributeCode: string, values: string[]) => void;
    toggleFilter: (attributeCode: string, value: string) => void;
    pageSize: number;
    setPageSize: (pageSize: number) => void;
};

export const useCategoryContent = (
    props: TUseCategoryContentProps
): TUseCategoryContent => {
    const {
        query,
        currentPage,
        categoryId,
        sortField,
        sortDir,
        setTotalPages
    } = props;

    const [activeCategory, setActiveCategory] = useState<number | null>(null);
    const {
        getFilterQuery,
        activeFilters,
        setFilter,
        toggleFilter,
        setActiveFilters
    } = useProductFilters();

    const handleToggleFilter = useCallback(
        (attributeCode: string, value: string) => {
            if (attributeCode === CATEGORY_ATTRIBUTE_CODE) {
                const values = activeFilters.get(attributeCode) || [];
                const valueIndex = values.indexOf(value);
                const newValues: string[] = [];

                if (valueIndex === -1) {
                    newValues.push(value);
                } else {
                    if (activeCategory) {
                        newValues.push(activeCategory.toString());
                    }
                }

                setFilter(attributeCode, newValues);
            } else {
                toggleFilter(attributeCode, value);
            }
        },
        [toggleFilter, setFilter, activeFilters, activeCategory]
    );

    const { pageSize, setPageSize } = usePageSize();
    const [data, setData] = useState<GetCategoryContentQuery | null>(null);
    const [runQuery, queryResponse] = useLazyQuery<
        GetCategoryContentQuery,
        GetCategoryContentQueryVariables
    >(query, {
        fetchPolicy: fetchPolicy.queries.default,
        onCompleted: data => {
            setData(data);
        }
    });

    // Run the category query immediately and whenever its variable values change.
    useEffect(() => {
        runQuery({
            variables: {
                currentPage: Number(currentPage),
                id: String(activeCategory),
                onServer: false,
                pageSize: Number(pageSize),
                sort: {
                    [sortField]: sortDir
                },
                // @ts-expect-error
                filter: getFilterQuery()
            }
        });
    }, [
        currentPage,
        activeCategory,
        pageSize,
        sortField,
        sortDir,
        runQuery,
        getFilterQuery,
        activeFilters,
        setActiveFilters
    ]);

    const { loading, error } = queryResponse;
    const [firstCategory] = filterOutNullableValues(data?.categoryList);

    const category = firstCategory;
    const products = filterOutNullableValues(
        data?.products?.items
    ) as unknown as TProduct[];
    const filters = filterOutNullableValues(data?.products?.aggregations);

    const totalPages = data?.products?.page_info?.total_pages ?? 0;
    const totalCount = data?.products?.total_count ?? 0;

    useEffect(() => {
        setTotalPages(totalPages);
    }, [setTotalPages, totalPages]);

    useEffect(() => {
        setActiveCategory(categoryId);
        setActiveFilters(
            new Map([[CATEGORY_ATTRIBUTE_CODE, [categoryId.toString()]]])
        );
    }, [categoryId, setActiveCategory, setActiveFilters]);

    return {
        loading,
        error: error?.message,
        category,
        products,
        filters,
        totalCount,
        activeFilters,
        setFilter,
        toggleFilter: handleToggleFilter,
        pageSize,
        setPageSize
    };
};
