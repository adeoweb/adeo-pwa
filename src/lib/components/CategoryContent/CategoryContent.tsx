import React, { Fragment, FunctionComponent, useEffect } from 'react';
import Pagination from 'src/lib/components/Pagination';
import { Col, Container, Row } from 'react-bootstrap';
import { ICategoryContentProps } from 'src/lib/components/CategoryContent';
import ProductGrid from 'src/lib/components/ProductGrid';
import ProductList from 'src/lib/components/ProductList';
import RichText from 'src/lib/components/RichText';
import ProductResults from 'src/lib/components/ProductResults';
import ProductsSort from 'src/lib/components/ProductsSort';
import LayoutModes from 'src/lib/components/LayoutModes';
import { LAYOUT_MODE_GRID } from 'src/lib/constants/layoutModes';
import LoadingIndicator, {
    FullPageLoadingIndicator
} from '../../components/LoadingIndicator';
import Breadcrumbs from 'src/lib/components/Breadcrumbs';
import { useAppContext } from 'src/peregrine/lib/context/adeoweb/app';
import ProductsSidebar from 'src/lib/components/ProductsSidebar';
import ProductFilters from 'src/lib/components/ProductFilters';
import { useMessageCard } from 'src/peregrine/lib/talons/adeoweb/MessageCard/useMessageCard';
import { useWishlist } from 'src/peregrine/lib/talons/adeoweb/Wishlist/useWishlist';
import MessageType from 'src/lib/constants/message';

const CategoryContent: FunctionComponent<ICategoryContentProps> = ({
    categoryId,
    category,
    pageControl,
    pageSize,
    sortControl,
    products,
    totalCount,
    filterControl,
    isLoading
}) => {
    const { filters, activeFilters, setFilter, toggleFilter } = filterControl;
    const description = category && category.description;
    const { currentPage, setPage, totalPages } = pageControl;
    const totalProducts = totalCount;
    const [{ layoutMode }] = useAppContext();
    const { addMessage, clearAllMessages } = useMessageCard();
    const {
        isAddingToWishlist,
        addToWishlistError,
        isRemovingFromWishlist,
        removeFromWishlistError
    } = useWishlist();

    useEffect(() => {
        if (addToWishlistError || removeFromWishlistError) {
            clearAllMessages();
            addMessage({
                type: MessageType.ERROR,
                message: String(addToWishlistError || removeFromWishlistError)
            });
        }
    }, [
        addMessage,
        clearAllMessages,
        addToWishlistError,
        removeFromWishlistError
    ]);

    return (
        <Fragment>
            {(isAddingToWishlist || isRemovingFromWishlist) && (
                <LoadingIndicator />
            )}
            {isLoading && <FullPageLoadingIndicator />}
            <Breadcrumbs categoryId={categoryId} />
            <Container>
                <Row>
                    <Col lg={9}>
                        {description && <RichText content={description} />}
                        <nav className="toolbox">
                            <div className="toolbox-left">
                                <ProductsSort sortControl={sortControl} />
                            </div>
                            <ProductResults
                                totalProducts={totalProducts}
                                pageSize={pageSize}
                                currentPage={currentPage}
                            />
                            <LayoutModes />
                        </nav>
                        {layoutMode === LAYOUT_MODE_GRID ? (
                            <ProductGrid items={products} category={category} />
                        ) : (
                            <ProductList items={products} category={category} />
                        )}
                        {totalPages > 1 && (
                            <nav className="toolbox toolbox-pagination">
                                <ProductResults
                                    totalProducts={totalProducts}
                                    pageSize={pageSize}
                                    currentPage={currentPage}
                                />
                                <Pagination
                                    currentPage={currentPage}
                                    setPage={setPage}
                                    totalPages={totalPages}
                                />
                            </nav>
                        )}
                    </Col>
                    <ProductsSidebar>
                        <ProductFilters
                            filters={filters}
                            activeFilters={activeFilters}
                            setFilter={setFilter}
                            toggleFilter={toggleFilter}
                        />
                    </ProductsSidebar>
                </Row>
            </Container>
            <div className="mb-5" />
        </Fragment>
    );
};

export default CategoryContent;
