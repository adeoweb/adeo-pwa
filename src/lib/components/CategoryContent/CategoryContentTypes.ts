import { IProductsSortControl } from 'src/lib/components/ProductsSort';
import { TCategoryInterface } from 'src/lib/types/graphql/Category';
import { TProduct } from 'src/lib/types/graphql/Product';
import { TAggregation } from 'src/lib/types/graphql/Aggregation';

export interface ICategoryPageControl {
    currentPage: number;
    setPage: (page: number) => void;
    totalPages: number;
}

export interface ICategoryFilterControl {
    filters: TAggregation[];
    activeFilters: Map<string, string[]>;
    setFilter: (attributeCode: string, values: string[]) => void;
    toggleFilter: (attributeCode: string, value: string) => void;
}

export interface ICategoryContentProps {
    categoryId: number;
    category: TCategoryInterface;
    pageControl: ICategoryPageControl;
    pageSize: number;
    sortControl: IProductsSortControl;
    products: TProduct[];
    totalCount: number;
    filterControl: ICategoryFilterControl;
    isLoading: boolean;
}
