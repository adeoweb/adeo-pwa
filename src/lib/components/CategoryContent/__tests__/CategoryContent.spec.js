import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import testRenderer from 'react-test-renderer';
import CategoryContent from 'src/lib/components/CategoryContent';
import { CategoryContentPropsMock } from 'src/lib/components/CategoryContent/__mocks__/CategoryContent.mock';

jest.mock('src/lib/components/Breadcrumbs', () => () => null);
jest.mock('src/lib/components/AddToWishlist', () => () => null);

jest.mock('src/peregrine/lib/talons/adeoweb/Product/useAddToCart', () => {
    return {
        useAddToCart: () => ({
            handleAddToCart: () => {},
            isAddToCartDisabled: false
        })
    };
});

jest.mock('src/peregrine/lib/talons/adeoweb/Product/useProductCompare', () => {
    return {
        useProductCompare: () => ({
            setProductHandler: () => {},
            isProductBeingCompared: () => false
        })
    };
});

jest.mock('src/peregrine/lib/context/adeoweb/app', () => {
    const state = {
        layoutMode: 'grid'
    };
    const api = {};
    const useAppContext = jest.fn(() => [state, api]);

    return { useAppContext };
});

jest.mock('src/peregrine/lib/talons/adeoweb/LayoutModes/useLayoutModes', () => {
    return {
        useLayoutModes: jest.fn(() => ({
            layoutMode: 'grid',
            setGrid: jest.fn(() => {}),
            setList: jest.fn(() => {})
        }))
    };
});

jest.mock('src/peregrine/lib/talons/adeoweb/MessageCard/useMessageCard', () => {
    const useMessageCard = jest.fn(() => ({
        addMessage: jest.fn(() => {}),
        clearAllMessages: jest.fn(() => {})
    }));

    return { useMessageCard };
});

jest.mock('src/peregrine/lib/talons/adeoweb/Wishlist/useWishlist', () => {
    const useWishlist = jest.fn(() => ({
        isAddingToWishlist: false,
        addToWishlistError: null,
        isRemovingFromWishlist: false,
        removeFromWishlistError: null
    }));

    return { useWishlist };
});

test('Renders correctly', () => {
    const component = testRenderer.create(
        <Router>
            <CategoryContent {...CategoryContentPropsMock} />
        </Router>
    );
    expect(component.toJSON()).toMatchSnapshot();
});
