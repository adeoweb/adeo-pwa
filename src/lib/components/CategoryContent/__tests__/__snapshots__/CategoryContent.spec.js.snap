// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
Array [
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col-lg-9"
      >
        <div
          dangerouslySetInnerHTML={
            Object {
              "__html": "description",
            }
          }
          onClick={[Function]}
        />
        <nav
          className="toolbox"
        >
          <div
            className="toolbox-left"
          >
            <div
              className="toolbox-item toolbox-sort"
            >
              <form
                className=""
                noValidate={true}
              >
                <div
                  className="select-custom"
                >
                  <select
                    className="form-control"
                    name="orderby"
                    onChange={[Function]}
                    value="relevance"
                  >
                    <option
                      value="relevance"
                    >
                      Sort by popularity
                    </option>
                    <option
                      value="position"
                    >
                      Sort by position
                    </option>
                    <option
                      value="name"
                    >
                      Sort by name
                    </option>
                    <option
                      value="price"
                    >
                      Sort by price
                    </option>
                  </select>
                </div>
              </form>
              <button
                className="btn btn-link sorter-btn btn-desc"
                onClick={[Function]}
                title="Set Ascending Direction"
              >
                <span
                  className="sr-only"
                >
                  Set Ascending Direction
                </span>
              </button>
            </div>
          </div>
          <div
            className="toolbox-item toolbox-show"
          >
            <label>
              Showing
               
              1
              –
              20
               
              of
               
              50
               
              results
            </label>
          </div>
          <div
            className="layout-modes"
          >
            <button
              className="layout-btn btn-grid btn btn-link p-0 active"
              onClick={[MockFunction]}
              title="Grid"
            >
              <i
                className="icon-mode-grid"
              />
            </button>
            <button
              className="layout-btn btn-grid btn btn-link p-0 "
              onClick={[MockFunction]}
              title="List"
            >
              <i
                className="icon-mode-list"
              />
            </button>
          </div>
        </nav>
        <div
          className="row-sm row"
        >
          <div
            className="col-xl-3 col-md-4 col-6"
          >
            <div
              className="product-default mb-4"
            >
              <figure>
                <a
                  href="/accessories/strive-shoulder-pack.html"
                  onClick={[Function]}
                >
                  <div
                    className="undefined undefined"
                  >
                    <img
                      alt="Image"
                      className="undefined undefined"
                      height={300}
                      loading="eager"
                      src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAQAAAAFCAQAAADIpIVQAAAADklEQVR42mNkgAJGIhgAALQABsHyMOcAAAAASUVORK5CYII="
                      style={
                        Object {
                          "height": "300px",
                          "width": "280px",
                        }
                      }
                      width={280}
                    />
                    <img
                      alt="Image"
                      className="undefined undefined"
                      height={300}
                      loading="lazy"
                      onError={[Function]}
                      onLoad={[Function]}
                      src="https://stg.lemona.lt/media/catalog/product/cache/1bb5596fe10909d977e0c8185b18ce46/m/b/mb04-black-0.jpg"
                      style={
                        Object {
                          "height": "300px",
                          "width": "280px",
                        }
                      }
                      width={280}
                    />
                  </div>
                </a>
              </figure>
              <div
                className="product-details"
              >
                <h2
                  className="product-title"
                >
                  <a
                    href="/accessories/strive-shoulder-pack.html"
                    onClick={[Function]}
                  >
                    Strive Shoulder Pack
                  </a>
                </h2>
                <div
                  className="price-box"
                >
                  <span
                    className="product-price"
                  >
                    <div />
                  </span>
                </div>
                <div
                  className="product-action"
                >
                  <button
                    className="btn-icon btn-add-cart"
                    disabled={false}
                    onClick={[Function]}
                  >
                    <div
                      className="add-to-cart-content"
                    >
                      <svg
                        fill="none"
                        height="24"
                        viewBox="0 0 24 24"
                        width="24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M16.667 7.99954V6.03288C16.667 5.42004 16.5463 4.81321 16.3118 4.24702C16.0772 3.68084 15.7335 3.16639 15.3002 2.73305C14.8668 2.29971 14.3524 1.95596 13.7862 1.72144C13.22 1.48692 12.6132 1.36621 12.0003 1.36621C11.3875 1.36621 10.7807 1.48692 10.2145 1.72144C9.64828 1.95596 9.13383 2.29971 8.70049 2.73305C8.26715 3.16639 7.92341 3.68084 7.68889 4.24702C7.45437 4.81321 7.33366 5.42004 7.33366 6.03288V10.6995C7.33366 10.8764 7.4039 11.0459 7.52892 11.1709C7.65395 11.296 7.82351 11.3662 8.00033 11.3662C8.17714 11.3662 8.34671 11.296 8.47173 11.1709C8.59675 11.0459 8.66699 10.8764 8.66699 10.6995V9.33288H14.0003V7.99954H8.66699V6.03288C8.66699 5.14882 9.01818 4.30098 9.6433 3.67586C10.2684 3.05073 11.1163 2.69954 12.0003 2.69954C12.8844 2.69954 13.7322 3.05073 14.3573 3.67586C14.9825 4.30098 15.3337 5.14882 15.3337 6.03288V10.6662C15.3337 10.843 15.4039 11.0126 15.5289 11.1376C15.6539 11.2626 15.8235 11.3329 16.0003 11.3329C16.1771 11.3329 16.3467 11.2626 16.4717 11.1376C16.5968 11.0126 16.667 10.843 16.667 10.6662V9.33288H20.0003V21.3329H4.00033V9.33288H6.00033V7.99954H2.66699V21.3929C2.66699 21.7306 2.80115 22.0545 3.03994 22.2933C3.27874 22.5321 3.60262 22.6662 3.94033 22.6662H20.0603C20.398 22.6662 20.7219 22.5321 20.9607 22.2933C21.1995 22.0545 21.3337 21.7306 21.3337 21.3929V7.99954H16.667Z"
                          fill="#21293C"
                        />
                      </svg>
                      Add to cart
                    </div>
                  </button>
                  <button
                    className="paction add-compare"
                    onClick={[Function]}
                    title="Add to Compare"
                  >
                    <span>
                      Add to Compare
                    </span>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
        <nav
          className="toolbox toolbox-pagination"
        >
          <div
            className="toolbox-item toolbox-show"
          >
            <label>
              Showing
               
              1
              –
              20
               
              of
               
              50
               
              results
            </label>
          </div>
          <div
            className="pagination"
          >
            <li
              aria-label="move to the previous page"
              className="page-item disabled"
            >
              <a
                className="page-link page-link-btn"
                href="#"
                onClick={[Function]}
              >
                <i
                  className="icon-angle-left"
                />
              </a>
            </li>
            <li
              className="page-item active"
            >
              <a
                className="page-link"
                href="//?page=1"
                onClick={[Function]}
              >
                1
                <span
                  className="sr-only"
                >
                  (current)
                </span>
              </a>
            </li>
            <li
              className="page-item "
            >
              <a
                className="page-link"
                href="//?page=2"
                onClick={[Function]}
              >
                2
              </a>
            </li>
            <li
              className="page-item"
            >
              <span>
                ...
              </span>
            </li>
            <li
              className="page-item "
            >
              <a
                className="page-link"
                href="//?page=9"
                onClick={[Function]}
              >
                9
              </a>
            </li>
            <li
              aria-label="move to the next page"
              className="page-item "
            >
              <a
                className="page-link page-link-btn"
                href="//?page=2"
                onClick={[Function]}
              >
                <i
                  className="icon-angle-right"
                />
              </a>
            </li>
          </div>
        </nav>
      </div>
      <aside
        className="sidebar-shop col-lg-3 order-lg-first"
      >
        <div
          className="pin-wrapper"
        >
          <div
            className="sidebar-wrapper"
            style={
              Object {
                "borderBottom": "0px none rgb(118, 127, 132)",
              }
            }
          >
            <div
              className="widget"
            >
              <h3
                className="widget-title"
              >
                <button
                  aria-expanded={true}
                  className=""
                  onClick={[Function]}
                >
                  Category
                </button>
              </h3>
              <div
                className="collapse show"
              >
                <div
                  className="widget-body"
                >
                  <ul
                    className="cat-list"
                  >
                    <li>
                      <button
                        onClick={[Function]}
                      >
                        <i
                          className="icon-check-empty"
                        />
                        <span
                          dangerouslySetInnerHTML={
                            Object {
                              "__html": "New Luma Yoga Collection (13)",
                            }
                          }
                        />
                      </button>
                    </li>
                    <li>
                      <button
                        onClick={[Function]}
                      >
                        <i
                          className="icon-check-empty"
                        />
                        <span
                          dangerouslySetInnerHTML={
                            Object {
                              "__html": "Tops (50)",
                            }
                          }
                        />
                      </button>
                    </li>
                    <li>
                      <button
                        onClick={[Function]}
                      >
                        <i
                          className="icon-check-empty"
                        />
                        <span
                          dangerouslySetInnerHTML={
                            Object {
                              "__html": "Jackets (12)",
                            }
                          }
                        />
                      </button>
                    </li>
                    <li>
                      <button
                        onClick={[Function]}
                      >
                        <i
                          className="icon-check-empty"
                        />
                        <span
                          dangerouslySetInnerHTML={
                            Object {
                              "__html": "Hoodies & Sweatshirts (12)",
                            }
                          }
                        />
                      </button>
                    </li>
                    <li>
                      <button
                        onClick={[Function]}
                      >
                        <i
                          className="icon-check-empty"
                        />
                        <span
                          dangerouslySetInnerHTML={
                            Object {
                              "__html": "Tees (12)",
                            }
                          }
                        />
                      </button>
                    </li>
                    <li>
                      <button
                        onClick={[Function]}
                      >
                        <i
                          className="icon-check-empty"
                        />
                        <span
                          dangerouslySetInnerHTML={
                            Object {
                              "__html": "Bras & Tanks (14)",
                            }
                          }
                        />
                      </button>
                    </li>
                    <li>
                      <button
                        onClick={[Function]}
                      >
                        <i
                          className="icon-check-empty"
                        />
                        <span
                          dangerouslySetInnerHTML={
                            Object {
                              "__html": "Women Sale (12)",
                            }
                          }
                        />
                      </button>
                    </li>
                    <li>
                      <button
                        onClick={[Function]}
                      >
                        <i
                          className="icon-check-empty"
                        />
                        <span
                          dangerouslySetInnerHTML={
                            Object {
                              "__html": "Tees (12)",
                            }
                          }
                        />
                      </button>
                    </li>
                    <li>
                      <button
                        onClick={[Function]}
                      >
                        <i
                          className="icon-check-empty"
                        />
                        <span
                          dangerouslySetInnerHTML={
                            Object {
                              "__html": "Erin Recommends (11)",
                            }
                          }
                        />
                      </button>
                    </li>
                    <li>
                      <button
                        onClick={[Function]}
                      >
                        <i
                          className="icon-check-empty"
                        />
                        <span
                          dangerouslySetInnerHTML={
                            Object {
                              "__html": "Performance Fabrics (13)",
                            }
                          }
                        />
                      </button>
                    </li>
                    <li>
                      <button
                        onClick={[Function]}
                      >
                        <i
                          className="icon-check-empty"
                        />
                        <span
                          dangerouslySetInnerHTML={
                            Object {
                              "__html": "Eco Friendly (6)",
                            }
                          }
                        />
                      </button>
                    </li>
                  </ul>
                </div>
              </div>
            </div>
            <div
              className="widget"
            >
              <h3
                className="widget-title"
              >
                <button
                  aria-expanded={true}
                  className=""
                  onClick={[Function]}
                >
                  Color
                </button>
              </h3>
              <div
                className="collapse show"
              >
                <div
                  className="widget-body"
                >
                  <ul
                    className="config-swatch-list"
                  >
                    <li
                      className=""
                    >
                      <button
                        onClick={[Function]}
                      >
                        <span
                          className="color-panel"
                          style={
                            Object {
                              "backgroundColor": "black",
                            }
                          }
                        />
                        <span>
                          Black (12)
                        </span>
                      </button>
                    </li>
                    <li
                      className=""
                    >
                      <button
                        onClick={[Function]}
                      >
                        <span
                          className="color-panel"
                          style={
                            Object {
                              "backgroundColor": "blue",
                            }
                          }
                        />
                        <span>
                          Blue (22)
                        </span>
                      </button>
                    </li>
                    <li
                      className=""
                    >
                      <button
                        onClick={[Function]}
                      >
                        <span
                          className="color-panel"
                          style={
                            Object {
                              "backgroundColor": "brown",
                            }
                          }
                        />
                        <span>
                          Brown (1)
                        </span>
                      </button>
                    </li>
                    <li
                      className=""
                    >
                      <button
                        onClick={[Function]}
                      >
                        <span
                          className="color-panel"
                          style={
                            Object {
                              "backgroundColor": "gray",
                            }
                          }
                        />
                        <span>
                          Gray (8)
                        </span>
                      </button>
                    </li>
                    <li
                      className=""
                    >
                      <button
                        onClick={[Function]}
                      >
                        <span
                          className="color-panel"
                          style={
                            Object {
                              "backgroundColor": "green",
                            }
                          }
                        />
                        <span>
                          Green (18)
                        </span>
                      </button>
                    </li>
                    <li
                      className=""
                    >
                      <button
                        onClick={[Function]}
                      >
                        <span
                          className="color-panel"
                          style={
                            Object {
                              "backgroundColor": "orange",
                            }
                          }
                        />
                        <span>
                          Orange (21)
                        </span>
                      </button>
                    </li>
                    <li
                      className=""
                    >
                      <button
                        onClick={[Function]}
                      >
                        <span
                          className="color-panel"
                          style={
                            Object {
                              "backgroundColor": "purple",
                            }
                          }
                        />
                        <span>
                          Purple (23)
                        </span>
                      </button>
                    </li>
                    <li
                      className=""
                    >
                      <button
                        onClick={[Function]}
                      >
                        <span
                          className="color-panel"
                          style={
                            Object {
                              "backgroundColor": "red",
                            }
                          }
                        />
                        <span>
                          Red (18)
                        </span>
                      </button>
                    </li>
                    <li
                      className=""
                    >
                      <button
                        onClick={[Function]}
                      >
                        <span
                          className="color-panel"
                          style={
                            Object {
                              "backgroundColor": "white",
                            }
                          }
                        />
                        <span>
                          White (9)
                        </span>
                      </button>
                    </li>
                    <li
                      className=""
                    >
                      <button
                        onClick={[Function]}
                      >
                        <span
                          className="color-panel"
                          style={
                            Object {
                              "backgroundColor": "yellow",
                            }
                          }
                        />
                        <span>
                          Yellow (16)
                        </span>
                      </button>
                    </li>
                  </ul>
                </div>
              </div>
            </div>
            <div
              className="widget"
            >
              <h3
                className="widget-title"
              >
                <button
                  aria-expanded={true}
                  className=""
                  onClick={[Function]}
                >
                  Eco Collection
                </button>
              </h3>
              <div
                className="collapse show"
              >
                <div
                  className="widget-body"
                >
                  <ul
                    className="cat-list"
                  >
                    <li>
                      <button
                        onClick={[Function]}
                      >
                        <i
                          className="icon-check-empty"
                        />
                        No (38)
                      </button>
                    </li>
                    <li>
                      <button
                        onClick={[Function]}
                      >
                        <i
                          className="icon-check-empty"
                        />
                        Yes (12)
                      </button>
                    </li>
                  </ul>
                </div>
              </div>
            </div>
          </div>
        </div>
      </aside>
    </div>
  </div>,
  <div
    className="mb-5"
  />,
]
`;
